runtime/smithy-client/common/src/aws/smithy/kotlin/runtime/client/config/CompressionClientConfig.kt

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package aws.smithy.kotlin.runtime.client.config
+
+import aws.smithy.kotlin.runtime.InternalApi
+import aws.smithy.kotlin.runtime.compression.CompressionAlgorithm
+import aws.smithy.kotlin.runtime.compression.Gzip
+
+@DslMarker
+public annotation class CompressionClientConfigDsl
+
+/**
+ * The configuration properties for a client that supports compression
+ */
+public interface CompressionClientConfig {
+    public val requestCompression: RequestCompressionConfig
+
+    @CompressionClientConfigDsl
+    public interface Builder {
+        public var requestCompression: RequestCompressionConfig.Builder
+
+        public fun requestCompression(block: RequestCompressionConfig.Builder.() -> Unit) {
+            requestCompression.apply(block)
+        }
+    }
+}
+
+/**
+ * The configuration properties for request compression.
+ */
+public class RequestCompressionConfig(builder: Builder) {
```

Comment:
**fix**: constructor should be `internal`

Reply:
I think this needs to be pubic, it's used when converting the builder to a config instance `override val requestCompression: RequestCompressionConfig = RequestCompressionConfig(builder.requestCompression)`.